### ISSUE TITLE
Docker container does not create the database

### ISSUE THREAD

**Issue by tmontanaro** [ging/fiware-idm#93]
Hello,
if I try to run the IdM service with docker compose it does not work.
Specifically, it creates 2 containers (mysql and idm) but, if I look at the the IdM container logs I find a lot of this messages:
```
waiting for database connection
```
After investigating about the issues, I found the problem: the database is not completely created. If I execute a `show databases` in the mysql container I obtain only 4 databases:
* information_schema
* mysql
* performance_schema
* sys

(I partially solved the problem by using a (maybe, because there is no connection among the 2 repositories) newer version of the Idm, and specifically by following this tutorial:
[FIWARE/tutorials.Identity-Management](https://github.com/FIWARE/tutorials.Identity-Management)
)

**Comment by minimalisti** [comment]
I have had similar issues. I have found that using the docker-compose.yml from the Keyrock tutorial you linked to I still have to run some Node scripts inside a running Keyrock container. My notes about this:

### Setting up Keyrock database
After containers are up, additional steps are required inside Keyrock container. To open a terminal running in Keyrock container:
_$ docker exec -it basic_access_control_keyrock-idm_1 bash_
Your container name might be different from _basic_access_control_keyrock-idm_1_. You can use _docker ps_ command to see all running containers.

When you are in the terminal inside Keyrock-container, run the following commands:
first read the README.md inside the container: _$ cat README.md_
- create the default database structure (if the database exists,  script exits): _$ npm run-script create_db_
- creates the database structure: _$ npm run-script migrate_db_
- seed tables with the admin user, etc.: _$ npm run-script seed_db_

This hopefully this helps, even thou it doesn't resolve the issue of why the Keyrock database is not initialized.

**Comment by tmontanaro** [comment]
Unfortunately, it does not work.
However, I understood the problem: it is downloading the last version of the IdM container but is trying to using the old configuration.
I think that the issues could be solved by simply specifying the right version in the yaml

**Comment by tmontanaro** [comment]
In fact, by setting `image: fiware/idm:7.5.1` in the yaml, instead of image: ~~fiware/idm~~, it works without your @minimalisti workaround!
If possible, developers should simply specify the version in the yaml!

**Comment by jason-fox** [comment]
Between Keyrock 7.5.1 and Keyrock 7.6.0, the ENV variable identifying the database has changed:

```diff
-      - DATABASE_HOST=mysql-db
+      - IDM_DB_HOST=mysql-db
```

The `docker-compose` was using old ENV and hence no database was set. It was mentioned deep in the docs somewhere, but ideally there should have been more info when connecting to the DB.

**Comment by tmontanaro** [comment]
Solved by #95
